machineconfiguration/v1/types.go

```diff
@@ -760,3 +760,137 @@ type ContainerRuntimeConfigList struct {
 
 	Items []ContainerRuntimeConfig `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PinnedImageSet describes a set of images that should be pinned.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
```

Comment:
do we want this in tech preview? I think we should probably put this under v1alpha1

Reply:
I am not sure honestly we can fine-tune this when we get the API stable.

Reply:
Yeah let's revisit once we get the PR stable. v1alpha sounds reasonable.